src/sentry/services/hybrid_cloud/user/impl.py

```diff
@@ -174,22 +174,17 @@ def get_first_superuser(self) -> Optional[RpcUser]:
         return serialize_rpc_user(user)
 
     def get_or_create_user_by_email(
-        self, *, email: str, ident: Optional[str] = None, referrer: Optional[str] = None
-    ) -> RpcUser:
+        self,
+        *,
+        email: str,
+        ident: Optional[str] = None,
+        referrer: Optional[str] = None,
+        skip_creation: Optional[bool] = False,
+    ) -> Optional[RpcUser]:
```

Comment:
Isn't it odd that `get_or_create` has a parameter that makes it do something totally different? Should we have a `get_user_or_none` method instead?

Reply:
It is odd. If I get_or_none, then I have to create in a different rpc. I was trying to minimize number of rpc calls

Reply:
I can call it get_user_by_email and add a flag that says `should_create`. That sounds better.

Reply:
Oh got it , you're suggesting to have both.